controllers/platformoperator_controller.go

```diff
@@ -100,162 +59,32 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	log.Info("reconciling request", "req", req.NamespacedName)
 	defer log.Info("finished reconciling request", "req", req.NamespacedName)
 
+	// TODO: flesh out status condition management
 	po := &platformv1alpha1.PlatformOperator{}
 	if err := r.Get(ctx, req.NamespacedName, po); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
-
-	css := &operatorsv1alpha1.CatalogSourceList{}
-	if err := r.List(ctx, css); err != nil {
-		log.Error(err, "failed to list the catalogsource resources in the cluster")
-		return ctrl.Result{}, err
-	}
-	if len(css.Items) == 0 {
-		log.Info("unable to query catalog content as no catalogsources are available")
-		return ctrl.Result{}, nil
-	}
-
-	readySources := make([]operatorsv1alpha1.CatalogSource, 0, len(css.Items))
-	for _, cs := range css.Items {
-		if cs.Status.GRPCConnectionState == nil {
-			continue
-		}
-		if cs.Status.GRPCConnectionState.LastObservedState != "READY" {
-			continue
-		}
-		if cs.Status.GRPCConnectionState.Address == "" {
-			continue
-		}
-		readySources = append(readySources, cs)
-	}
-	log.Info("filtering out bundles into candidates", "package name", po.Spec.PackageName, "channel name", channelName)
-	var (
-		errors []error
-		cb     candidateBundles
-	)
-	// TODO: Should build a cache for efficiency
-	for _, cs := range readySources {
-		// Note(tflannag): Need to account for grpc-based CatalogSource(s) that
-		// specify a spec.Address or a spec.Image, so ensure this field exists, and
-		// it's not empty before creating a registry client.
-		rc, err := registryClient.NewClient(cs.Status.GRPCConnectionState.Address)
-		if err != nil {
-			errors = append(errors, fmt.Errorf("failed to register client from the %s/%s grpc connection: %w", cs.GetName(), cs.GetNamespace(), err))
-			continue
-		}
-		it, err := rc.ListBundles(ctx)
-		if err != nil {
-			errors = append(errors, fmt.Errorf("failed to list bundles from the %s/%s catalog: %w", cs.GetName(), cs.GetNamespace(), err))
-			continue
-		}
-		for b := it.Next(); b != nil; b = it.Next() {
-			if b.PackageName != po.Spec.PackageName || b.ChannelName != channelName {
-				continue
-			}
-			cb = append(cb, b)
-		}
-	}
-	if len(errors) != 0 {
-		agg := utilerror.NewAggregate(errors)
-		log.Error(agg, "failed to build catalog snapshot")
-		return ctrl.Result{}, agg
+	desiredBundle, err := r.Sourcer.Source(ctx, po)
+	if err != nil {
+		log.Error(err, "failed to source content for platform operator")
```

Comment:
nit: should we include the name of the PO in a formatted string in the error log for more visibility? 
```
log.Error(err, "failed to source content for platform operator", po.GetName())
```

Reply:
We should already have this information from the logger given we create a logger from the Reconcile ctx provided to us: `log := logr.FromContext(ctx)`.